src/pages/instances/CreateInstanceForm.tsx

```diff
@@ -245,97 +251,111 @@ const CreateInstanceForm: FC = () => {
   }
 
   return (
-    <main className="l-main">
-      <div className="p-panel">
-        <div className="p-panel__header">
-          <h1 className="p-panel__title">Create an instance</h1>
-        </div>
-        <div className="p-panel__content create-instance">
-          <Form onSubmit={() => submit(formik.values)} stacked className="form">
-            <InstanceFormMenu
-              active={section}
-              setActive={updateSection}
-              isConfigDisabled={!formik.values.image}
-              isConfigOpen={isConfigOpen}
-              toggleConfigOpen={toggleMenu}
-            />
-            <Row className="form-contents" key={section}>
-              <Col size={12}>
-                <NotificationRow />
-                {section === INSTANCE_DETAILS && (
-                  <InstanceCreateDetailsForm
-                    formik={formik}
-                    project={project}
-                    onSelectImage={handleSelectImage}
-                  />
-                )}
+    <>
+      {isProjectLoading && <Loader />}
+      {!projectExists && !isProjectLoading && <NoMatch />}
+      {projectExists && (
```

Comment:
Those three lines are repeated in too many components. I think we should have a wrapper (like `ProtectedRoute`) that handles this. Interface could be `<ProtectedRoute outlet={<ProjectLoader outlet={<InstanceList />}} />` in `App.tsx`. Or  you find a more elegant solution.